### PR TITLE
brings back barticles for bonfires

### PR DESCRIPTION
**Pull request by Fikou**
Reverts tgstation/tgstation#59916
BARTICLES BONFIRES BATUMS

### CODE REVIEW

**Comment by Kylerace** [comment]
this is only on basil right now so no mergies

File: code/game/objects/structures/bonfire.dm [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+/obj/structure/bonfire/buckle_mob(mob/living/buckled_mob, force = FALSE, check_loc = TRUE)
+	if(..())
+		buckled_mob.pixel_y += 13
+
+/obj/structure/bonfire/unbuckle_mob(mob/living/buckled_mob, force=FALSE)
+	if(..())
+		buckled_mob.pixel_y -= 13
+
+/particles/bonfire
```
Comment (tralezab) [inline code/game/objects/structures/bonfire.dm:164]: FYI:  we will need to make a new system for particles. Currently, only one particle per object is enforced, so we need dummy objects to be linked to particles, and those dummy objects to follow the actual object we want particles for

File: code/game/objects/structures/bonfire.dm [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	lifespan = 0.7 SECONDS
+	fade = 1 SECONDS
+	grow = -0.01
+	velocity = list(0, 0)
+	position = generator("circle", 0, 16, NORMAL_RAND)
+	drift = generator("vector", list(0, -0.2), list(0, 0.2))
+	gravity = list(0, 0.95)
+	scale = generator("vector", list(0.3, 0.3), list(1,1), NORMAL_RAND)
+	rotation = 30
+	spin = generator("num", -20, 20)
```
Comment (tralezab) [inline code/game/objects/structures/bonfire.dm:180]: also, look at all the code above. an in game particle tester for coders would be monumental

**Comment by Cyberboss** [comment]
Don't draft PRs you're not the author of @MrStonedOne

**Comment by Mothblocks** [comment]
its past the 28th. we can merge now

**Comment by MrStonedOne** [comment]
@Cyberboss 
> Don't draft PRs you're not the author of @MrStonedOne

If it has the dnm label im gonna draft it.

**Comment by Fikou** [comment]
@Mothblocks

**Comment by Mothblocks** [comment]
i'm forcing 514 like a boss

**Pull request merged**